components/brave_welcome_ui/components/background/index.tsx

```diff
@@ -55,43 +63,65 @@ function AnimatedBackground (props: { children?: JSX.Element }) {
       .to(stars04, { transform: 'scale(1.5)' })
 
     setScenes({ s1, s2 })
-  }, [])
+  }, [isReadyForAnimation])
```

Comment:
No, the content should not use `isReadyForAnimation`. 

We check for `ref.current` inside the body of useEffect. If the handle of ref is null then it won't setup WAAPI objects.

`<S.Box ref={isReadyForAnimation ? ref : null}>`

Reply:
Perhaps that wasn't clear. I should revise this add both checks in one condition. 

`if (!ref.current && isReadyForAnimation) { ... }`

Reply:
Well either way it needs a comment. But yes it seems clearer to have all the logic for that in 1 place. The `ref` attribute doesn't need to know whether animation will be used on that ref, so maybe it's clearer to do make that decision based on `isReadyForAnimation` in in this effect... otherwise we're splitting the logic....

The other option you could consider, which may be better is not using `React.useRef` for this and instead using `React.useCallback((ref) => {...` and passing that to the ref prop in the JSX. That will avoid having to use a ref but sync an effect on a prop. You would perform the contents of both effects in the callback when the incoming `ref` param is truthy.

Reply:
added anther check. an easier route.